src/infrastructure/content/ContentCommunicator.java

```diff
@@ -1,24 +1,143 @@
+	 */
+	private HashMap<String, IContentNotificationHandler> mapHandler = new HashMap<String, IContentNotificationHandler>();
+	
+	/**
+	 * Getting the communicator of networking module to send data over the network
+	 */
+	private ICommunicator communicator = CommunicatorFactory.getCommunicator(port);
+	
+	/**
+	 * This method provides userName to other classes within this package
+	 * @return userName - username of the client
+	 */
+	protected String getUserName() {
+		return userName;
+	}
+	
+	/**
+	 * This method provides HashMap mapImage to other classes within this package
+	 * @return mapImage - The HashMap which stores the userName and userImage
+	 */
+	protected HashMap<String, String> getMapImage() {
+		return mapImage;
+	}
+	
+	/**
+	 * This method allows other classes within this package to set the HashMap mapImage
+	 * @param mapImage - The HashMap which stores the userName and userImage of all clients
+	 */
+	protected void setMapImage(HashMap<String, String> mapImage) {
+		this.mapImage = mapImage;
+	}
+	
+	/**
+	 * This method provides HashMap mapHandler to other classes within this package
+	 * @return mapHandler - The HashMap which stores the identifier and handler of subscriber of this module
+	 */
+	protected HashMap<String, IContentNotificationHandler> getMapHandler() {
+		return mapHandler;
+	}
+	
+	/**
+	 * This method initializes a user by storing its username and userimage in a map as Strings
+	 * This method stores these values locally too and generates server's ipAddress by getting
+	 * 		the port value of the board from ServerPort class
+	 * Finally, some manipulation on JSON string is done and then sent to server via networking
+	 * @param userDetails - This is a JSON string which contains a JSON object with three fields, 
+	 * 		namely ipAddress, username and image.
+	 */
 	public void initialiseUser(String userDetails) {
-    	
-    }
+		JSONObject jsonObject = new JSONObject(userDetails);
+		serverIPAddress = jsonObject.getString("ipAddress");
+		userName = jsonObject.getString("username");
+		userImage = jsonObject.getString("image");
+		ServerPort serverPort = new ServerPort();
+		port = serverPort.getPort();
+		serverIPAddress += ":";
+		serverIPAddress += String.valueOf(port);
+		mapImage.put(userName, userImage);
+		jsonObject.put("meta", "newUser");
+		jsonObject.remove("ipAddress");
+		communicator.send(serverIPAddress, jsonObject.toString(), "contentServer");
+	}
 	
-	@Override
+	/**
+	 * This method gets the message from UI module and then send it via networking module
+	 * A meta field is added to it indicating that it is message before sending to server
+	 * @param message - This is a JSON string which contains a JSON object with key "message"
+	 */
 	public void sendMessageToContent(String message) {
-    	
-    }
+		JSONObject jsonObject = new JSONObject(message);
+		jsonObject.put("meta", "message");
+		communicator.send(serverIPAddress, jsonObject.toString(), "contentServer");

```

Comment:
Why are you using a networking communicator object to send message to content? I don't understand what is going on.

Reply:
How else can I do it? I have to send messages to networking.